### ISSUE TITLE
[P1] Real iPEPS excitation basis drops imaginary effective-Hamiltonian matrix elements

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#956]
> 🤖 **AI-generated issue** - written by Codex from the requested iPEPS code review.

`_make_basis` creates float arrays, so the gradients used by `_build_H_and_N` differentiate only real components of B. Imaginary off-diagonal matrix elements are lost even when the energy is an exact complex Hermitian quadratic form. This is an independent assembly defect that would remain after fixing mixed-RDM normalization and axis order.

Sources: [real basis allocation](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_excitations.py#L514-L525), [gradient-based matrix assembly](https://github.com/tenax-lab/tenax/blob/42aef1f1a94e1ca8400f6187400aa3613e3e175e/src/tenax/algorithms/ipeps_excitations.py#L566-L588).

## Isolated Reproduction

This assembly-level probe substitutes exact quadratic functionals in memory to isolate matrix assembly from the other contraction defects. It does not modify source files.

```python
import jax.numpy as jnp
from tenax.algorithms.ipeps_config import CTMEnvironment
import tenax.algorithms.ipeps_excitations as ex

env = CTMEnvironment(
    *([jnp.ones((1, 1))] * 4 + [jnp.ones((1, 1, 1))] * 4)
)
import numpy as np
from unittest.mock import patch

A = jnp.array([1., 0.]).reshape(1, 1, 1, 1, 2)
M = jnp.array([[1., -1j], [1j, 1.]])
def energy(A, B, *args):
    return jnp.real(jnp.vdot(B.ravel(), M @ B.ravel()))
def norm(A, B, *args):
    return jnp.real(jnp.vdot(B, B))
with patch.object(ex, "_compute_excitation_energy", energy), \
     patch.object(ex, "_compute_norm", norm):
    H, N = ex._build_H_and_N(
        A, env, jnp.zeros(2), jnp.zeros((2, 2, 2, 2)),
        0., 2, ex.ExcitationConfig(num_excitations=2),
    )
print("H:", H, "N:", N)
print("computed:", ex._solve_excitations(H, N, 2))
print("expected:", np.linalg.eigvalsh(np.asarray(M)))
```

Actual H=2I, N=2I and eigenvalues `[1, 1]`; expected eigenvalues `[0, 2]`. The common factor of two is not the issue: the imaginary entries carrying the spectral splitting disappear.

## Suggested Fix

Differentiate complex B with the appropriate conversion from JAX's complex cotangent convention, or differentiate independent real and imaginary coordinates and reconstruct the complex matrices. Merely changing the basis dtype is insufficient without checking conjugation/factor conventions. Add exact complex Hermitian quadratic-form reconstruction and spectrum tests, plus a physical complex-state/gate excitation case.

Reproduced on commit `42aef1f1a94e1ca8400f6187400aa3613e3e175e` using `JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 uv run python`. No repository code was modified for these probes.
